File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        l, r = 0, len(skill) - 1
        psum = skill[l] + skill[r]
        tsum = 0
        while l < r:
            sum_pair = skill[l] + skill[r]
            if psum != sum_pair:
                return -1
            tsum += skill[l] * skill[r]
            psum = sum_pair
            l += 1
            r -= 1

        return tsum
        

File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(),skill.end());
        int l = 0, r = skill.size()-1;
        long long int sum,psum=skill[l]+skill[r],tsum=0;
        while(l<r){
            sum=skill[l]+skill[r];
            
            if(psum!=sum){
                return -1;
            }
            tsum+=skill[l]*skill[r];
            psum=sum;
            l++;
            r--;
        }
        return tsum;
    }
};